src/main/java/org/spongepowered/common/data/processor/data/TargetedLocationDataProcessor.java

```diff
@@ -0,0 +1,136 @@
+            final TargetedLocationData newData = checkNotNull(overlap.merge(checkNotNull(manipulator), data));
+            final Location location = newData.target().get();
+            return Optional.of(manipulator.set(Keys.TARGETED_LOCATION, location));
+        }
+        return Optional.absent();    }
+
+    @Override
+    public Optional<TargetedLocationData> fill(DataContainer container, TargetedLocationData targetedLocationData) {
+        final Location location = DataUtil.getData(container, Keys.TARGETED_LOCATION, Location.class);
+        return Optional.of(targetedLocationData.set(Keys.TARGETED_LOCATION, location));
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, TargetedLocationData manipulator) {
+        return DataTransactionBuilder.failResult(manipulator.getValues());
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, TargetedLocationData manipulator, MergeFunction function) {
+        return DataTransactionBuilder.failResult(manipulator.getValues());
+    }
+
+    @Override
+    public Optional<ImmutableTargetedLocationData> with(Key<? extends BaseValue<?>> key, Object value, ImmutableTargetedLocationData immutable) {
+        if (key == Keys.TARGETED_LOCATION) {
+            return Optional.<ImmutableTargetedLocationData>of(new ImmutableSpongeTargetedLocationData(immutable.target().get()));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public DataTransactionResult remove(DataHolder dataHolder) {
+        if (dataHolder instanceof EntityLiving) {
+            final DataTransactionBuilder builder = DataTransactionBuilder.builder();
+            final Optional<TargetedLocationData> optional = from(dataHolder);
+            if (optional.isPresent()) {
+                try {
+                    EntityLiving living = (EntityLiving) dataHolder;
+                    living.getNavigator().clearPathEntity();
+                    return builder.replace(optional.get().getValues()).result(DataTransactionResult.Type.SUCCESS).build();
+                } catch (Exception e) {
+                    Sponge.getLogger().error("There was an issue resetting the target location on an entity!", e);
+                    return builder.result(DataTransactionResult.Type.ERROR).build();
+                }
+            } else {
+                return builder.result(DataTransactionResult.Type.SUCCESS).build();
+            }
+        }
+        return DataTransactionBuilder.failNoData();
+    }
+
+    @Override
+    public TargetedLocationData create() {
+        return null; // new SpongeTargetedLocationData() ?
+    }
+
+    @Override
+    public ImmutableTargetedLocationData createImmutable() {
+        return null; // new ImmutableSpongeTargetedLocationData() ?
+    }
+
+    @Override
+    public Optional<TargetedLocationData> createFrom(DataHolder dataHolder) {
+        return from(dataHolder); //something more?
```

Comment:
You always want to return a DataMapiulator here if the data holder is applicable. The difference between this method and `from` is that while `from` can return absent when passed a supported data holder (if no pathfinding path is set, for example), this method shouldn't.